src/core/currency/wallet/currency-wallet-api.ts

```diff
@@ -560,6 +561,31 @@ export function makeCurrencyWalletApi(
       await engine.saveTx(tx)
       fakeCallbacks.onTransactionsChanged([tx])
     },
+    async saveTxAction(
+      txid: string,
+      tokenId: string | null,
+      savedAction: EdgeTxAction
+    ): Promise<void> {
+      const { accountApi } = input.props.output.accounts[accountId]
+      const { allTokens, currencyInfo } = accountApi.currencyConfig[pluginId]
+
+      const { currencyCode } = upgradeCurrencyCode({
+        allTokens,
+        currencyInfo,
+        currencyCode: undefined,
+        tokenId: tokenId ?? undefined
+      })
```

Comment:
Try `const { currencyCode } = tokenId == null ? currencyInfo : allTokens[tokenId]`. It's shorter and simpler.

The `upgradeCurrencyCode` function is designed to deal with backwards-compatibility issues, and since this is brand new code, it seems weird to use it.